src/storage/mvcc/txn.rs

```diff
@@ -185,8 +331,14 @@ impl<S: Snapshot> MvccTxn<S> {
                         ttl: lock.ttl,
                     });
                 }
-                // No need to overwrite the lock and data.
-                // If we use single delete, we can't put a key multiple times.
+                if lock.lock_type == LockType::Pessimistic {
```

Comment:
I think we don't need to check this.
TiDB already has a unit test that makes sure prewrite request for pessimistic transaction sets `IsPessimisticLock` field.

Reply:
never mind it doesn't hurt for a redundant check.

Reply:
Maybe we could remove it in the future PR